### PR TITLE
Fix OsuLegacyDecoder's parser being NULL if no ruleset was specified

### PR DESCRIPTION
**Pull request by marco-stephan**
Fixes https://github.com/ppy/osu/issues/1076

### CODE REVIEW

File: osu.Game/Beatmaps/Formats/OsuLegacyDecoder.cs [hunk 1]
```diff
@@ -456,6 +456,11 @@ protected override void ParseFile(StreamReader stream, Beatmap beatmap)
                         handleColours(beatmap, line, ref hasCustomColours);
                         break;
                     case Section.HitObjects:
+
+                        // If the ruleset wasn't specified, assume the osu!standard ruleset.
+                        if(parser == null)
```
Comment (peppy) [inline osu.Game/Beatmaps/Formats/OsuLegacyDecoder.cs]: space between `if` and `(`

**Comment by RemieRichards** [comment]
Wouldn't this be better in handleGeneral()? where the other parsers are made?

**Comment by MrTheMake** [comment]
It is not really possible though as `handleGeneral(...)` isn't called for the mode.

**Comment by RemieRichards** [comment]
But there's the mode switch() statement in handleGeneral(), that depending on the mode value makes the correct parser.

**Comment by MrTheMake** [comment]
This function is only called for a "general line" and "mode" is one of those. If a beatmap doesn't contain this line, the `handleGeneral(...)` isn't called for the mode line and therefore the switch doesn't switch to mode. The whole issue was that the file doesn't contain this line.

**Comment by RemieRichards** [comment]
Ah, shit, that makes sense.

**Review by peppy — APPROVED** [review]

**Pull request merged**